src/app/store/store.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { GroceriesService } from '../groceries.service';
+import { Item } from '../models/item';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-store',
+  templateUrl: './store.component.html',
+  styleUrls: ['./store.component.css'],
+})
+export class StoreComponent implements OnInit, OnDestroy {
+  filter: string = 'all';
+  sorting: string = 'dont';
+  store: Item[] = [];
+  groceriesServiceSubscription!: Subscription;
+
+  constructor(private readonly groceriesService: GroceriesService) {}
+
+  ngOnInit(): void {
+    this.groceriesServiceSubscription = this.groceriesService
```

Comment:
Remember we need to store the subscription and unsubscribe from it in order to prevent memory leaks